### PR TITLE
[Bugfix] SSO creating a new account  instead of update the existing one with the given email

### PR DESCRIPTION
**Pull request by AguilarKevin**
image

Context: existing account(created using credentials) couldn't access using the SSO implementation because this plugin only search the users by their azure id and not their email

### CODE REVIEW

**Review by mitch-pfm — COMMENTED** [review]
Changes look good to me but I will wait for the review from @joebuonocore before merging.

**Review by jgreutman-pfm — APPROVED** [review]
Good work!

**Review by joebuonocore — COMMENTED** [review]
@AguilarKevin Looks great! cc: @mitch-pfm

**Pull request merged**
